wsd/Admin.cpp

```diff
@@ -603,42 +603,32 @@ unsigned Admin::getNetStatsInterval()
 
 std::string Admin::getChannelLogLevels()
 {
-    std::string result;
-    // Get the list of channels..
-    std::vector<std::string> nameList;
-    Log::logger().names(nameList);
+    unsigned int wsdLogLevel = Log::logger().get("wsd").getLevel();
+    std::string result = "wsd=" + levelList[wsdLogLevel];
 
-    std::string levelList[9] = {"none", "fatal", "critical", "error", "warning", "notice", "information", "debug", "trace"};
-
-    for (size_t i = 0; i < nameList.size(); i++)
-    {
-        result += (nameList[i] != "" ? nameList[i]: "?") + '=' + levelList[Log::logger().get(nameList[i]).getLevel()] + (i != nameList.size() - 1 ? " ": "");
-    }
+    result += " kits=" + (_forkitLogLevel.empty() != true ? _forkitLogLevel: levelList[wsdLogLevel]);
 
     return result;
 }
 
-void Admin::setChannelLogLevel(const std::string& _channelName, std::string _level)
+void Admin::setChannelLogLevel(const std::string& _channelName, std::string level)
 {
     assertCorrectThread();
 
-    std::string levelList[9] = {"none", "fatal", "critical", "error", "warning", "notice", "information", "debug", "trace"};
-    if (std::find(std::begin(levelList), std::end(levelList), _level) == std::end(levelList))
-    {
-        _level = "trace";
-    }
-
     // Get the list of channels..
     std::vector<std::string> nameList;
     Log::logger().names(nameList);
 
-    for (size_t i = 0; i < nameList.size(); i++)
+    if (std::find(std::begin(levelList), std::end(levelList), level) == std::end(levelList))
+        level = "trace";
```

Comment:
Should we really default to trace? I expect information (or debug, at most). Trace is only used for troubleshooting and not recommended as a default. See loolwsd.xml for the default config value please.

Reply:
It is a fallback rather than default. It should never get hit. I changed it to debug.